python/lsst/cp/pipe/ptc/cpExtractPtcTask.py

```diff
@@ -449,18 +425,88 @@ def measureMeanVarCov(self, exposure1, exposure2, region=None, covAstierRealSpac
         wDiff = np.where(diffIm.getMask().getArray() == 0, 1, 0)
         w = w12*wDiff
 
-        if np.sum(w) < self.config.minNumberGoodPixelsForFft:
+        if np.sum(w) < self.config.minNumberGoodPixelsForFastFourierTransform:
             self.log.warn(f"Number of good points for FFT ({np.sum(w)}) is less than threshold "
-                          f"({self.config.minNumberGoodPixelsForFft})")
+                          f"({self.config.minNumberGoodPixelsForFastFourierTransform})")
             return np.nan, np.nan, None
 
         maxRangeCov = self.config.maximumRangeCovariancesAstier
         if covAstierRealSpace:
+            # Calculate  covariances in real space.
             covDiffAstier = computeCovDirect(diffIm.image.array, w, maxRangeCov)
         else:
-            shapeDiff = diffIm.image.array.shape
-            fftShape = (fftSize(shapeDiff[0] + maxRangeCov), fftSize(shapeDiff[1]+maxRangeCov))
-            c = CovFft(diffIm.image.array, w, fftShape, maxRangeCov)
-            covDiffAstier = c.reportCovFft(maxRangeCov)
+            # Calculate covariances via FFT (default).
+            shapeDiff = np.array(diffIm.image.array.shape)
+            # Calculate the sizes of FFT dimensions.
+            s = shapeDiff + maxRangeCov
+            tempSize = np.array(np.log(s)/np.log(2.)).astype(int)
+            fftSize = np.array(2**(tempSize+1)).astype(int)
+            fftShape = (fftSize[0], fftSize[1])
+
+            c = CovFastFourierTransform(diffIm.image.array, w, fftShape, maxRangeCov)
+            covDiffAstier = c.reportCovFastFourierTransform(maxRangeCov)
 
         return mu, varDiff, covDiffAstier
+
+    def fillPtcDatasetAmp(self, ptcDataset, ampName, rawExpTime=[np.nan], rawMean=[np.nan], rawVar=[np.nan],
```

Comment:
This method makes me think there maybe should be a method (or set of methods) on the PTC object to assist in construction.  This sets everything to single-element lists for each amplifier property, correct?  I think my idea is an `initAmp` method that accepts lists like this is doing, but that then checks that all the lengths for supplied values match.  This could then be used here as well as in `CpSolvePtcTask` when the dataset is filled with the collated results, and also even in the `PtcDataset.__init__`.  Maybe on the next pass at refactoring.

Reply:
I have placed the amp-filling function as part of the ptcDataset code in`ip_isr`.